src/devices/CharacterLcd/Hd44780.cs

```diff
@@ -68,181 +77,312 @@ public class Hd44780 : Hd44780Base, IDisposable
         /// </summary>
-        public override bool BacklightOn
+        /// <param name="microseconds">Time to wait if checking busy state isn't possible/practical.</param>
+        protected void WaitForNotBusy(int microseconds)
         {
-            get
-            {
-                return _backlight != -1 && _controller.Read(_backlight) == PinValue.High;
-            }
-            set
-            {
-                if (_backlight != -1)
-                    _controller.Write(_backlight, value ? PinValue.High : PinValue.Low);
-            }
+            _interface.WaitForNotBusy(microseconds);
         }
 
         /// <summary>
-        /// Send a data or command byte to the controller.
+        /// Clears the LCD, returning the cursor to home and unshifting if shifted.
+        /// Will also set to Increment.
         /// </summary>
-        /// <param name="data">True to send data, otherwise sends a command.</param>
-        protected override void Send(byte value, bool data = false)
+        public void Clear()
         {
-            _controller.Write(_rsPin, data ? PinValue.High : PinValue.Low);
+            SendCommand(ClearDisplayCommand);
+            WaitForNotBusy(3000);
+        }
 
-            if (_rwPin != -1)
-            {
-                _controller.Write(_rwPin, PinValue.Low);
-            }
+        /// <summary>
+        /// Moves the cursor to the first line and first column.
+        /// </summary>
+        public void Home()
+        {
+            SendCommand(ReturnHomeCommand);
+            WaitForNotBusy(3000);
+        }
 
-            if (_dataPins.Length == 8)
-            {
-                WriteBits(value, 8);
-            }
-            else
-            {
-                WriteBits((byte)(value >> 4), 4);
-                WriteBits(value, 4);
-            }
+        /// <summary>
+        /// Moves the cursor to an explicit column and row position.
+        /// </summary>
+        /// <param name="left">The column position from left to right starting with 0.</param>
+        /// <param name="top">The row position from the top starting with 0.</param>
+        public void SetCursorPosition(int left, int top)
+        {
+            int rows = _rowOffsets.Length;
+            if (top < 0 || top >= rows)
+                throw new ArgumentOutOfRangeException(nameof(top));
+
+            int newAddress = left + _rowOffsets[top];
+            if (left < 0 || (rows == 1 && newAddress >= 80) || (rows > 1 && newAddress >= 104))
+                throw new ArgumentOutOfRangeException(nameof(left));
+
+            SendCommand((byte)(SetDDRamAddressCommand | newAddress));
+        }
+
+        /// <summary>
+        /// Enable/disable the display.
+        /// </summary>
+        public bool DisplayOn
+        {
+            get => (_displayControl & DisplayControl.DisplayOn) > 0;
+            set => SendCommand((byte)(value ? _displayControl |= DisplayControl.DisplayOn
```

Comment:
consider using `[Flags]` enum and HasFlag and similar utilities for these

Reply:
DisplayControl is a `[Flags]` enum. `.HasFlag` had terrible [perf](https://blogs.msdn.microsoft.com/dotnet/2018/04/18/performance-improvements-in-net-core-2-1/) before 2.1 so I generally avoid it. I also find it weird as we don't have `SetFlag` or `ClearFlag`, so I lean towards doing everything with operators for consistency.